pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -238,6 +238,8 @@ func (cb *ClusterBuilder) buildSubsetCluster(opts buildClusterOpts, destRule *co
 func (cb *ClusterBuilder) applyDestinationRule(mc *MutableCluster, clusterMode ClusterMode, service *model.Service,
 	port *model.Port, proxyNetworkView map[network.ID]bool, destRule *config.Config, serviceAccounts []string) []*cluster.Cluster {
 	destinationRule := CastDestinationRule(destRule)
+
+	host := service.Hostname.String()
```

Comment:
I think this is not right for dns resolution, think about ServiceEntry with DNS type.

Reply:
I added a logging statement and applied a ServiceEntry with `resolution: DNS` and it was printing out the expected hostnames individually (for multiple hosts this still works). Is there anything I am missing? If this is not right for DNS resolution, do you have any helpful resources I could use to choose the hostname?
```
apiVersion: networking.istio.io/v1alpha3
kind: ServiceEntry
metadata:
  name: badssl-service-entry
  namespace: default
spec:
  hosts:
  - badssl.com
  - self-signed.badssl.com
  location: MESH_EXTERNAL
  ports:
  - number: 443
    name: https
    protocol: TLS
  resolution: DNS
```

Reply:
Forexample

···
// apiVersion: networking.istio.io/v1alpha3
// kind: ServiceEntry
// metadata:
//   name: external-svc-dns
// spec:
//   hosts:
//   - foo.bar.com
//   location: MESH_EXTERNAL
//   ports:
//   - number: 80
//     name: http
//     protocol: HTTP
//   resolution: DNS
//   endpoints:
//   - address: us.foo.bar.com
//     ports:
//       http: 8080
//   - address: uk.foo.bar.com
//     ports:
//       http: 9080
//   - address: in.foo.bar.com
//     ports:
//       http: 7080
···